src/DotLiquid/Util/StrFTime.cs

```diff
@@ -124,17 +150,28 @@ public static string ToStrFTime(this DateTime dateTime, string format, CultureIn
         /// <param name="culture">the CurrentCulture to be used when formatting</param>
         /// <returns>a string version of date-time matching pattern.</returns>
         public static string ToStrFTime(this DateTimeOffset dateTimeOffset, string format, CultureInfo culture)
+            => ObjectToStrFTime(dateTimeOffset, format, culture);
+
+        private static string ObjectToStrFTime(object source, string format, CultureInfo culture)
         {
-            culture = culture ?? throw new ArgumentException(message: "CultureInfo is mandatory", paramName: "culture");
-            return Regex.Replace(input: format, pattern: SPECIFIER_REGEX,
-                evaluator: specifier => SpecifierEvaluator(
-                    specifier: specifier.Groups[0].Value,
-                    flags: specifier.Groups[GROUP_FLAGS].Captures.Cast<Capture>().Select(capture => capture.Value).ToList(),
-                    width: specifier.Groups[GROUP_WIDTH].Captures.Cast<Capture>().Select(capture => (int?)Convert.ToInt32(capture.Value)).FirstOrDefault(),
-                    directive: specifier.Groups[GROUP_DIRECTIVE].Captures.Cast<Capture>().Select(capture => capture.Value).FirstOrDefault(),
-                    source: dateTimeOffset,
-                    culture: culture
-                    ));
+            culture = culture ?? throw new ArgumentException(message: "CultureInfo is mandatory", paramName: nameof(culture));
+
+            try
+            {
+                return Regex.Replace(input: format, pattern: SPECIFIER_REGEX,
+                    evaluator: specifier => SpecifierEvaluator(
+                        specifier: specifier.Groups[0].Value,
+                        flags: specifier.Groups[GROUP_FLAGS].Captures.Cast<Capture>().Select(capture => capture.Value).ToList(),
+                        width: specifier.Groups[GROUP_WIDTH].Captures.Cast<Capture>().Select(capture => (int?)Convert.ToInt32(capture.Value)).FirstOrDefault(),
+                        directive: specifier.Groups[GROUP_DIRECTIVE].Captures.Cast<Capture>().Select(capture => capture.Value).FirstOrDefault(),
+                        source: source,
+                        culture: culture
+                        ));
+            }
+            catch (FormatException ex) when (ex.Message == "Only")

```

Comment:
Exception should by design by localized, such as programmatic string equality checks are dangerous. Is there any other programmatic check we could rely on for this filtering?

Reply:
This error is thrown in two places in Liquid not by .NET itself. We could make a custom internal exception, but seems like a lot of boilerplate code for something internal.

Reply:
If this is a local exception then absolutely we should be using our own exception definition. It is a common best practice to define own exception types and avoid throwing generic system exceptions. But, it is also a breaking change if that exception was not only internal but also sometimes thrown to the caller of our APIs. So proceed with caution.

Reply:
It is thrown on https://github.com/dotliquid/dotliquid/pull/559/files#diff-b7bfc6e3a7d66bf9e9f2047c1695a75add8ea64a5c7afd95630ea7b651c762e9R204 (strftime.cs line 66 and 204). I'm not clear what the expectation is - in those areas we don't have scope to format the message so we throw an exception, catch where we do have more info and properly format the message.